### PR TITLE
Create Account functionality

### PR DESCRIPTION
**Pull request by kelvinscuesta**
- Added Redux Thunk
- New Users hit with the front end component, to the database
- clean up in a few files
- need to fix dispatch method, keeps hitting the dispatch fail on catch

### CODE REVIEW

File: client/actions/actions.js [hunk 1]
```diff
@@ -1,6 +1,57 @@
...
+const successfulCreateAccount = user => ({
+  type: types.SIGN_UP_SUCCESS,
+  isFetching: false,
+  isAuthenticated: true,
+  id_token: user.id.token,
+});
+
+const failedCreateAccount = message => ({
+  type: types.SIGN_UP_FAIL,
+  isFetching: false,
```
Comment (nabramow) [inline client/actions/actions.js:28]: I believe isFetching is unnecessary when using Redux Thunk as thunk handles the asynchronicity.

File: client/components/createUser.jsx [hunk 2]
```diff
@@ -0,0 +1,86 @@
...
+    this.handleInputChange = this.handleInputChange.bind(this);
+    this.submitLogin = this.submitLogin.bind(this);
+  }
+
+  submitLogin(event) {
+    event.preventDefault();
+    const { username, password, email } = this.state;
+    const credentials = {
+      username,
+      password,
```
Comment (nabramow) [inline client/components/createUser.jsx:37]: Is this storing the hashed password in state or the unhashed?

File: server/server.js [hunk 3]
```diff
@@ -1,12 +1,14 @@
 const path = require('path');
 const express = require('express');
 require('dotenv').config();
-
+const cors = require('cors');
```
Comment (sergiyalariki) [inline server/server.js:4]: We are not using them

File: package.json [hunk 4]
```diff
@@ -42,17 +42,19 @@
...
     "style-loader": "^1.1.3",
     "webpack": "^4.42.0",
     "webpack-cli": "^3.3.11",
-    "webpack-dev-server": "^3.10.3",
-    "node-sass": "^4.13.1"
+    "webpack-dev-server": "^3.10.3"
   },
   "dependencies": {
     "bcrypt": "^4.0.1",
+    "cors": "^2.8.5",
```
Comment (sergiyalariki) [inline package.json:57]: Uninstall

**Review by nabramow — APPROVED** [review]
Made a couple comments, but otherwise looks good!

**Review by sergiyalariki — APPROVED** [review]

**Pull request merged**